src/TasksModule/Components/TasksBoard/TasksBoard.tsx

```diff
@@ -0,0 +1,230 @@
+import React, { useContext, useEffect, useState } from "react";
+import styles from "./TasksBoard.module.css";
+import axios from "axios";
+import { baseUrl } from "../../../Constants/Components/Urls";
+import { motion } from "framer-motion";
+import { AuthContext } from "../../../Context/Components/AuthContext.js";
+
+
+export default function TasksBoard() {
+  const token = localStorage.getItem("userToken");
+  const {userRole}:string=useContext(AuthContext);
+  const [toDoTasks, setToDoTasks] = useState<object[]>();
+  // interface TaskData{
+  //   id:number,
+  //   newstatus:string,
+  //   oldstatus:string
+  // }
+  const [inProgressTasks, setInProgressTasks] = useState<object[]>();
+  const [doneTasks, setDoneTasks] = useState<object[]>();
+  const toDo: object[] = [];
+  const inProgress: object[] = [];
+  const done: object[] = [];
+  // const [isDraggingOver,setIsDraggingOver]=useState<boolean>(false);
+
+  const getUserTasks = async () => {
+    try {
+      const response = await axios.get(
+        `${baseUrl}/Task?pageSize=100&pageNumber=1`,
+        {
+          headers: { Authorization: token },
+        }
+      );
+      console.log("userTasks ", response.data.data);
+      setTasks(response.data.data);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const setTasks = (tasks: object[]) => {
+    tasks?.map((task: object) => {
+      if (task?.status == "ToDo") toDo.push(task);
+      else if (task?.status == "InProgress") inProgress.push(task);
+      else done.push(task);
+    });
+    console.log("to do ", toDo);
+    setToDoTasks(toDo);
+    setInProgressTasks(inProgress);
+    setDoneTasks(done);
+  };
+
+  const changeTaskStatus = (e, newstatus: string) => {
+    const id = e.dataTransfer.getData("taskId");
+    const oldstatus: string = e.dataTransfer.getData("curruntStatus");
+    // console.log(id,newstatus,oldstatus);
+
+    if (newstatus != oldstatus) {
+      try {
+        const response = axios.put(
```

Comment:
Here we are storing a Promise in the response, we need to await until the promise is resolved using the await keyword. and we need to make this function asynchronous.